kDrive/AppDelegate.swift

```diff
@@ -82,6 +83,8 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, AccountManagerDeleg
         let currentState = RootViewControllerState.getCurrentState()
         prepareRootViewController(currentState: currentState)
 
+        (accountManager as? AccountManager)?.delegate = self
```

Comment:
If I don't cast, i get a crash in the DI. Do you know why @adrien-coye ?

Reply:
You cannot declare weak properties in a pure swift protocol (even if inherits from `AnyObject`)
You probably want to stick to a `setDelegate(_ )` method for this case.

You can declare a property weak in a protocol only if @Objc which we probably do not want.